src/lib/components/molecules/table/index.jsx

```diff
@@ -28,33 +30,42 @@ export function Table({ columns, data }) {
     })
   }
 
+  styles = mergeStyles(defaultStyles, styles)
+
   return (
-    <table class="w-full table-fixed">
+    <table className={styles.table}>
       <thead>
         <tr>
           {table.columns.map((column, index) => (
-            <th key={column.id} className={column.headerCellClass}>
-              <HeaderCell key={index} onClick={() => sortByColumn(index)} {...column.headerProps} />
+            <th key={column.id} className={mergeStyles(styles.headerCell, column.styles?.headerCell)}>
+              <HeaderCell
```

Comment:
I have to say - i find these three different lines relating to adding CSS quite confusing: 33, 36 and 40. Is there any way to remove the `mergeStyles()` parts. Or is that essential?

Reply:
Fair point. `mergeStyles()` is just a helper function that allows for overriding the styles, using the `styles` prop on the component or the column definition (see line 74 in `table.stories.jsx` for instance).

I agree that it would be better if we could automate this and remove all the boilerplate code. This is possible if we hook into the JSX compilation process, but that's somewhat complex, so I didn't want to take that on right away.

The choice was between this somewhat naïve solution for styled components, and using a library like [Emotion](https://emotion.sh/docs/introduction) or [Styled Components](https://styled-components.com/), which introduce a lot of new concepts. I felt it would be better for adoption and onboarding to stay as close as possible to how we currently write CSS. In future we should try and replicate some of the clever JSX magic that these libraries inject though.

That's my thinking, but I'm open to other suggestions.